### PR TITLE
[Merged by Bors] - feat(FiberedCategory/Cocartesian): define cocartesian & strongly cocartesian morphisms

### PR DESCRIPTION
**Pull request by callesonne**
This PR adds cocartesian & strongly cocartesian morphisms, and provides some basic API. This is just the opposite versions of cartesian & strongly cartesian morphisms, and the code is just copied from that file with the arrows reversed.

---

- [x] depends on: #13410

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/FiberedCategory/Cocartesian.lean [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+
+import Mathlib.CategoryTheory.FiberedCategory.HomLift
+
+/-!
+# Cocartesian morphisms
+
+This file defines cartesian resp. strongly cartesian morphisms with respect to a functor
+`p : 𝒳 ⥤ 𝒮`.
+
+All definitions in this file are opposites of the results in `FiberedCategory/Cartesian`.
```
Comment (dagurtomas) [inline Mathlib/CategoryTheory/FiberedCategory/Cocartesian.lean:15]: You should add a comment in the other file about keeping these two files in sync.

File: Mathlib/CategoryTheory/FiberedCategory/Cocartesian.lean [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+    [IsHomLift p g ψ'] (h : φ ≫ ψ = φ ≫ ψ') : ψ = ψ' := by
+  rw [map_uniq p f φ (g := g) rfl (φ ≫ ψ) ψ rfl, map_uniq p f φ (g := g) rfl (φ ≫ ψ) ψ' h.symm]
+
+@[simp]
+lemma map_self : map p f φ (comp_id f).symm φ = 𝟙 b := by
+  subst_hom_lift p f φ; symm
+  apply map_uniq
+  simp only [comp_id]
+
+/-- The composition of two `IsCocartesian.map` is also given by `IsCocartesian.map`. -/
```
Comment (dagurtomas) [inline Mathlib/CategoryTheory/FiberedCategory/Cocartesian.lean:223]: Could you add diagrams to this docstring like in the other file?

**Review by dagurtomas — COMMENTED** [review]
It would be nice to have some diagrams explaining the basic definitions in both files. Otherwise seems like a straightforward dualization of the other file, LGTM.

File: Mathlib/CategoryTheory/FiberedCategory/Cocartesian.lean (hunk 2 above)
Comment (callesonne) [reply Mathlib/CategoryTheory/FiberedCategory/Cocartesian.lean:233]: I added them for strongly cartesian / cocartesian, for cartesian / cocartesian I just added a text explanation as it did not seem much more helpful with a plaintext diagram.

File: Mathlib/CategoryTheory/FiberedCategory/Cocartesian.lean (hunk 1 above)
Comment (callesonne) [reply Mathlib/CategoryTheory/FiberedCategory/Cocartesian.lean]: Yes, thanks!

**Comment by dagurtomas** [comment]
Thanks!

maintainer merge

**Comment by semorrison** [comment]
bors merge

**Comment by jcommelin** [comment]
bors merge

**Comment by mattrobball** [comment]
bors merge

**Pull request closed without merging**